### PR TITLE
Update v_ksm_data_mart.sql

### PR DESCRIPTION
**Pull request by ssocite**
New modifications to the giving piece of the data mart views.

### CODE REVIEW

File: views/v_ksm_data_mart.sql [hunk 1]
```diff
@@ -581,86 +581,143 @@ Create or Replace View v_datamart_giving as
...
- or cal.curr_fy = fiscal_year + 5
- Then hh.FISCAL_YEAR Else 0 End) as giving_last_5_year
+from rpt_pbh634.v_ksm_giving_trans_hh
+--- Drop individual gifts with anonymous type 1, 2, 3 for the purpose of calculating the suggested fields�or alternatively drop all such records
+where trim(rpt_pbh634.v_ksm_giving_trans_hh.ANONYMOUS) is null),
+
+give as (select give.ID_NUMBER, 
+give.af_status,
+rpt_pbh634.ksm_pkg.get_fiscal_year(give.LAST_GIFT_DATE) as last_gift_fy
+from RPT_PBH634.v_Ksm_Giving_Summary give), --- stewardship amount
```
Comment (phively) [inline views/v_ksm_data_mart.sql:606]: This does not follow the anonymous special handling rules used in the hh subquery. Remove v_ksm_giving_summary and use hh instead, e.g.
rpt_pbh634.get_fiscal_year(max(hh.date_of_record)) as last_gift_fy
on line 605 (not tested)

File: views/v_ksm_data_mart.sql [hunk 2]
```diff
@@ -581,86 +581,143 @@ Create or Replace View v_datamart_giving as
...
+,KLC_Final.klc_fy_count as KSM_KLC_Total_FY_Count
+--- KLC Total Years in the Last 5
+,KLC_Final.klc_fy_count_5 as KSM_KLC_Last5_PFY_Count
+FROM degrees
+Left Join a on a.id_number = degrees.ID_NUMBER
+Left Join give on give.id_number = degrees.ID_NUMBER
+Left Join g on g.id_number = degrees.ID_NUMBER
+Left Join KLC_Final on KLC_Final.GIFT_CLUB_ID_NUMBER = degrees.ID_NUMBER
+Left Join Donor_Ind on Donor_Ind.id_number = degrees.id_number
+where a.ANONYMOUS_DONOR is null
```
Comment (phively) [inline views/v_ksm_data_mart.sql:723]: Let's also redact anyone whose spouse is null. Easiest way would just be to join on spouse_id_number e.g.
Left Join a a_spouse On a_spouse.spouse_id_number = degrees.id_number
and also add another condition
where a_spouse.anonymous_donor is null

**Review by phively — CHANGES_REQUESTED** [review]
-- Looks like the ellipses … in the comments on lines 587 and 600 are causing issues – you can replace with a comma
-- Update anonymous filtering: also drop if spouse is fully anonymous! No need to have logic for individual gifts, just the "a" subquery - easiest would be to left join on spouse_id_number
-- LAST_GIFT_FY needs to be fixed; should use the hh subquery, not call directly from v_ksm_giving_summary, to adhere to the anonymous gift logic

**Comment by ssocite** [comment]
Modifications to the data mart file and the mobile view file.

File: views/v_ksm_data_mart.sql [hunk 3]
```diff
@@ -581,86 +581,143 @@ Create or Replace View v_datamart_giving as
 
 
 
---- anonymous subquery 
+/* 
+EXCLUSIONS:
+1) Drop all records linked to an entity with the anonymous special handling code
+2) Drop individual gifts with anonymous type 1, 2, 3 for the purpose of calculating the suggested fields�or alternatively drop all such records
```
[the remark this answers is not in the document]
Comment (ssocite) [reply views/v_ksm_data_mart.sql]: comma added

File: views/v_ksm_data_mart.sql [hunk 4]
```diff
@@ -581,86 +581,143 @@ Create or Replace View v_datamart_giving as
...
---- Count of Fiscal Years of Giving (Current FY - Previous 5 FY)
-count(distinct Case When cal.curr_fy = fiscal_year 
- or cal.curr_fy = fiscal_year + 1
- or cal.curr_fy = fiscal_year + 2
- or cal.curr_fy = fiscal_year + 3
- or cal.curr_fy = fiscal_year + 4
- or cal.curr_fy = fiscal_year + 5
- Then hh.FISCAL_YEAR Else 0 End) as giving_last_5_year
+from rpt_pbh634.v_ksm_giving_trans_hh
+--- Drop individual gifts with anonymous type 1, 2, 3 for the purpose of calculating the suggested fields�or alternatively drop all such records
```
[the remark this answers is not in the document]
Comment (ssocite) [reply views/v_ksm_data_mart.sql]: comma added

**Pull request merged**